napari/_app_model/actions/_toggle_action.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from app_model.types import Action, ToggleRule
+
+if TYPE_CHECKING:
+    from ...viewer import Viewer
+
+
+class ViewerToggleAction(Action):
+    """Action subclass that toggles a boolean viewer (sub)attribute on trigger.
+
+    Parameters
+    ----------
+    id : CommandId
+        The id of the action.
+    title : str
+        The title of the action.
+    viewer_attribute : str
+        The attribute of the viewer to toggle. (e.g. 'axes')
+    sub_attribute : str
+        The attribute of the viewer attribute to toggle. (e.g. 'visible')
+    **kwargs
+        Additional keyword arguments to pass to the Action constructor.
+
+    Examples
+    --------
+    >>> action = ViewerToggleAction(
+    ...     id='some.command.id',
+    ...     title='Toggle Axis Visibility',
+    ...     viewer_attribute='axes',
+    ...     sub_attribute='visible',
+    ... )
+    """
+
+    def __init__(
+        self,
+        *,
+        id: str,
```

Comment:
Should this be 
```suggestion
        id: CommandId,
```
?

Reply:
`str` looks right. It's that way up through the pydantic model `CommandRule`.

Reply:
I will amend the docstring above to:
```python
    id : str
        The command id of the action.
```
@nclack let me know if this is not right.

Reply:
`id` ends up getting passed to the superclass `Action`. I'd follow whatever convention is used for documenting `Action`.

Reply:
From: https://app-model.readthedocs.io/en/latest/types/?h=action#app_model.types.Action

![image](https://user-images.githubusercontent.com/23182829/194198209-b591461c-02e0-44b3-b378-aad458b70f24.png)

So keep as is or change description to: 'A global identifier for the command.' (maybe to nitpicky)